cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/DiagnosticAnnotation.java

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io.ballerina.cli.utils;
+
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Represents a diagnostic annotation that is used to annotate the source code with diagnostics.
+ *
+ * @since 2201.9.0
+ */
+public class DiagnosticAnnotation {
+
+    private final ArrayList<String> lines;
+    private final int start;
+    private final int length;
+    private final boolean isMultiline;
+    private final int endOffset;
+    private final int startLineNumber;
+    private final int terminalWidth;
+    private final DiagnosticSeverity severity;
+    private final DiagnosticAnnotationType type;
+
+    public static final HashMap<DiagnosticSeverity, String> SEVERITY_COLORS = new HashMap<>() {{
+        put(DiagnosticSeverity.INTERNAL, "blue");
+        put(DiagnosticSeverity.HINT, "blue");
+        put(DiagnosticSeverity.INFO, "blue");
+        put(DiagnosticSeverity.WARNING, "yellow");
```

Comment:
Lets move this string values into a constant variables.
it will improve the modifiability of the code.